### PR TITLE
layout update

### PR DESCRIPTION
**Pull request by chinodelmundo**
updated main layout
- header
- sidebar

### CODE REVIEW

**Comment by davidallenby** [comment]
I'd advise to follow the Core, Features, Shared folder architecture. As this _might_ get unmanageable as the project scales.

**Comment by SchleichAlex** [comment]
Why did you fork the repository instead of creating a new branch on the existing repository for the PR?

**Comment by SchleichAlex** [comment]
i think the header and the sidebar components should be put inside of the main-layout as subfolders as they are part of the main-layout

File: frontend/src/app/components/sidebar/sidebar.component.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<div class="card flex justify-content-center">
```
Comment (SchleichAlex) [inline frontend/src/app/components/sidebar/sidebar.component.html:1]: bootstrap is currently not in the project so we cant use the bootstrap utility classes.

File: frontend/src/app/components/sidebar/sidebar.component.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  <p-menu [model]="items">
+    <ng-template pTemplate="item" let-item>
+      <ng-container *ngIf="item.route; else elseBlock">
```
Comment (SchleichAlex) [inline frontend/src/app/components/sidebar/sidebar.component.html:4]: i would use the new control-flow syntax as it improves readability tremendously. 
https://angular.dev/guide/templates/control-flow

File: frontend/src/app/components/sidebar/sidebar.component.html [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        <a [routerLink]="item.route" class="p-menuitem-link">
+          <span [class]="item.icon"></span>
+          <span class="ml-2">{{ item.label }}</span>
+        </a>
+      </ng-container>
+      <ng-template #elseBlock>
+        <a [href]="item.url" class="p-menuitem-link">
+          <span [class]="item.icon"></span>
+          <span class="ml-2">{{ item.label }}</span>
```
Comment (SchleichAlex) [inline frontend/src/app/components/sidebar/sidebar.component.html:13]: bootstrap class. Does not work

File: frontend/src/app/components/sidebar/sidebar.component.scss [hunk 4]
```diff
@@ -0,0 +1,10 @@
+::ng-deep .p-submenu-header {
```
Comment (SchleichAlex) [inline frontend/src/app/components/sidebar/sidebar.component.scss:1]: I would avoid overriding prime-ng classes as long as possible and rely on the default behavior

And IF we really have to override some classes something then we should add :host to only override them in the context of the sidebar and not globally. (::ng-deep styles will "breakout" of the view encapsulation https://blog.angular-university.io/angular-host-context/)

If you want to do global overrides i would suggest to put them in the styles.scss of the project

File: frontend/angular.json [hunk 5]
```diff
@@ -22,19 +22,20 @@
...
                 "input": "public"
-              }
+              },
+              "./src/assets"
             ],
             "styles": [
-              "src/styles.scss"
+              "src/styles.scss",
+              "node_modules/primeng/resources/primeng.min.css",
+              "primeng/resources/themes/fluent-light/theme.css"
```
Comment (davidallenby) [inline frontend/angular.json:38]: If you're using PrimeNG, might be a better idea to use their SASS theme and bundle it in with the custom `src/styles`. 
Move the styles file into a subfolder under src. E.g. `src/styles/styles.scss`

Comment (chinodelmundo) [reply frontend/angular.json:38]: i'm unfamiliar with this, skipping it for now if it's ok.

**Comment by chinodelmundo** [comment]
Worked on the comment, except for 

> SASS theme and bundle it

File: frontend/src/app/components/sidebar/sidebar.component.html (hunk 2 above)
Comment (SchleichAlex) [reply frontend/src/app/components/sidebar/sidebar.component.html]: Maybe you want to do this as its a good opportunity to learn new angular stuff.

File: frontend/src/app/main-layout/sidebar/sidebar.component.html [hunk 6]
```diff
@@ -0,0 +1,17 @@
+<div>
+  <p-tieredMenu [model]="items">
+    <ng-template pTemplate="item" let-item let-hasSubmenu="hasSubmenu">
+      <ng-container *ngIf="item.route; else withoutRoute">
+        <a [routerLink]="item.route" [href]="item.href" class="p-menuitem-link">
+          <span class="ml-2">{{ item.label }}</span>
```
Comment (SchleichAlex) [inline frontend/src/app/main-layout/sidebar/sidebar.component.html:6]: ml-2 is a bootstrap class

Comment (chinodelmundo) [reply frontend/src/app/main-layout/sidebar/sidebar.component.html]: missed that one thanks!

**Comment by SchleichAlex** [comment]
For me its fine. Dont know how i can accept the PR via github

**Review by SchleichAlex — APPROVED** [review]

**Pull request merged**